backend/Posts/views.py

```diff
@@ -68,3 +68,13 @@ def get_queryset(self):
         if qs is None:
             raise NotFound("No Post with this ID was found")
         return qs
+
+# Edit a single post by its ID
+class PostEdit(generics.UpdateAPIView):
+    queryset = Post.objects.all()
```

Comment:
you can filter here so that you only look for posts that YOU made, and it will make any potential unauthorized edits easier to detect.

Reply:
Changing the queryset resulted in 404 errors for the tests rather than 403 errors, which doesn't accurately represent that the user trying to edit the post doesn't have the proper permissions. Here is an image of the failed pytest:

![image](https://github.com/user-attachments/assets/82b86d0b-e04b-4af5-90fd-5850ccf2b6dc)

Reply:
yeah i mean it's up to you if you want to leave it with the entire posts in the query set. you're potentially leaving access to that post rather then not allowing access at all. on the client side it shouldn't make a difference since we only see edit post on posts that we make, but again up to you i just thought it would make a lot of the logic in your serializer simpler. my apologizes